src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -346,5 +353,37 @@ private string GetEnvironmentContentNonWindows(int agentId, string agentName, IL
 
             return builder.ToString();
         }
+
+        private async Task<string> DumpEventLogs(DateTime jobStartTimeUtc, string diagFolder){
+            var builder = new StringBuilder();
+
+            string powerShellExe = HostContext.GetService<IPowerShellExeUtil>().GetPath();
+            string arguments = $@"Get-WinEvent -ListLog * | ForEach-Object {{ Get-WinEvent -ErrorAction SilentlyContinue -FilterHashtable @{{ LogName=$_.LogName; StartTime='{ jobStartTimeUtc }'; EndTime='{ DateTime.UtcNow }';}} }} | Format-List > { diagFolder }\EventViewer-{ jobStartTimeUtc.ToString("yyyyMMdd-HHmmss") }.log";
```

Comment:
One-line script here looks hard-to-read and modify - let's split it up a little bit.

Reply:
Done